scripts/workers/submission_worker.py

```diff
@@ -676,9 +676,29 @@ def process_submission_callback(body):
                 SUBMISSION_LOGS_PREFIX, body
             )
         )
+        submission_pk = json.loads(body)["submission_pk"]
         body = yaml.safe_load(body)
         body = dict((k, int(v)) for k, v in body.items())
-        process_submission_message(body)
+        pushgateway_endpoint = os.environ.get("PUSHGATEWAY_ENDPOINT")
+        response = requests.post(
+            "{endpoint}/metrics/job/{j}/instance/{i}/submission_pk/{s}".format(
+                endpoint=pushgateway_endpoint,
+                j="submission-worker",
+                i="pushgateway",
+                s=submission_pk,
+            ),
+            data="{k} {v}\n".format(k="submissions_unloaded_from_queue", v=1),
```

Comment:
We can't count the submission as processed as soon as it is read from the queue. This should be done only after submission is processed successfully or marked failed

Reply:
@Ram81 i thought we are monitoring the submissions currently in queue.. so shouldn't be decrease the counter just after message is unloaded from the queue?

Reply:
No, we are monitoring number of unprocessed submissions. If a submission is not marked success/failed it is not considered as processed